src/backend/base/langflow/components/tools/GleanSearchAPI.py

```diff
@@ -0,0 +1,121 @@
+import httpx
+
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urljoin
+
+from langchain_core.pydantic_v1 import BaseModel
+
+from langflow.base.langchain_utilities.model import LCToolComponent
+from langflow.inputs import SecretStrInput, StrInput, NestedDictInput, IntInput
+from langflow.field_typing import Tool
+from langflow.schema import Data
+
+
+class GleanSearchAPIComponent(LCToolComponent):
+    display_name = "Glean Search API"
+    description = "Call Glean Search API"
+    name = "GleanAPI"
+
+    inputs = [
+        StrInput(
+            name="glean_api_url",
+            display_name="Glean API URL",
+            required=True,
+        ),
+        SecretStrInput(name="glean_access_token", display_name="Glean Access Token", required=True),
+        StrInput(name="query", display_name="Query", required=True),
+        IntInput(name="page_size", display_name="Page Size", value=10),
+        StrInput(name="field_name", display_name="Field Name", required=False),
+        NestedDictInput(name="values", display_name="Values", required=False),
```

Comment:
maybe `is_list=True`?

Reply:
@nicoloboschi for this, i ended up changing it to take a NestedDictInput for the full request options body - there are a ton of options and it sounds like some people want them exposed... i'd love to do something elegant with is_list etc but because there's so many more options, i figured this will keep it general but also clean up the UI just slightly....